variables.go

```diff
@@ -121,6 +121,8 @@ func (e *Executor) getVariables(t *Task, call Call) (Vars, error) {
 	if err := merge(call.Vars); err != nil {
 		return nil, err
 	}
+	r := varReplacer{vars: result}
+	result = r.replaceVars(result)
```

Comment:
The way I have implemented things with this proposal, I am afraid you might get some slightly _unexpected_ behavior, although I am not sure if it's really a problem:

Taskvars.yml:
```
A = "{{.B}}!" // B not defined here.
```

Taskfile.yml:
```
default:
  vars:
    B: "aha"
  cmds:
    - echo "{{.A}}}"
```
`task default` would echo "aha!".

This basically means that no templates are resolved before _all_ variables are resolved. I don't really know if this is the desired behavior, but it's better than no variables getting compiled. I think personally, something along the lines of the following _might_ be more in line with what _I_ would expect:

1. At startup, Taskvars and environment variables are merged/resolved and compiled. These get's stored on the executor (when/if support for `set` is dropped). As long as `set` is needed, it _might_ be better to store a map of _environment variables_ in the Context. See *(1).
2. For each task, fetch task variables, override with _compiled_ call variables and executor level variables, and compile the remaining variables.
3. Retrieve new task with "ReplaceVariables" (although I think this method might need a better name to be honest; it does way more than replacing variables).

@andreynering, what's your opinion on how this should work?

*(1) To ensure better concurrent behavior when relying on the `set` keyword, It might be better to store a new _copy_ of the overwritten environment variables (as map) in the Context for each call to `set`, and use that context for any task running _after_ `set` in the dependency graph. I.e., ensure this always gives the same results, no matter wich order task-a and task-b ends up being called in:
```yaml
default:
  deps:
    - task-a
    - task-b
task-a:
  deps:
    - task: set-evar
      vars: {SET: "A"}
  cmds:
    - echo "$EVAR" // use context to ensure this always echo "A"
task-b:
  deps:
    - task: set-evar
      vars: {SET: "B"}
  cmds:
    - echo "$EVAR"  // use context to ensure this always echo "B"
set-evar:
  cmds:
    - echo {{.SET}}
  set: EVAR
```
is the only reason you might want to store this in
- For each task,